ibis/pyspark/compiler.py

```diff
@@ -343,110 +346,77 @@ def fn(col):
+    else:
+        return ~compile_any(
+            t, expr, scope, context=context, window=window, **kwargs
+        )
 
 
 @compiles(ops.All)
 def compile_all(t, expr, scope, context=None, **kwargs):
-    def fn(col):
-        if 'window' in kwargs:
-            window = kwargs['window']
-            return F.min(col).over(window)
-        else:
-            return F.min(col)
-
-    return compile_aggregator(t, expr, scope, fn, context, **kwargs)
+    return compile_aggregator(t, expr, scope, F.min, context, **kwargs)
 
 
 @compiles(ops.NotAll)
-def compile_notall(t, expr, scope, context=None, **kwargs):
-    def fn(col):
-        if 'window' in kwargs:
-            window = kwargs['window']
-            return ~F.min(col).over(window)
-        else:
+def compile_notall(t, expr, scope, *, context=None, window=None, **kwargs):
+    # See comments for opts.NotAny for reasoning for the if/else
+    if context is None:
+
+        def fn(col):
             return ~(F.min(col))
 
-    return compile_aggregator(t, expr, scope, fn, context, **kwargs)
+        return compile_aggregator(t, expr, scope, fn, context, **kwargs)
+    else:
+        return ~compile_all(
+            t, expr, scope, context=context, window=window, **kwargs
+        )
 
 
 @compiles(ops.Count)
 def compile_count(t, expr, scope, context=None, **kwargs):
-    def fn(col):
-        if 'window' in kwargs:
-            window = kwargs['window']
-            return F.count(col).over(window)
-        else:
-            return F.count(col)
-
-    return compile_aggregator(t, expr, scope, fn, context, **kwargs)
+    return compile_aggregator(t, expr, scope, F.count, context, **kwargs)
 
 
 @compiles(ops.Max)
+@compiles(ops.CumulativeMax)
 def compile_max(t, expr, scope, context=None, **kwargs):
-    def fn(col):
-        if 'window' in kwargs:
-            window = kwargs['window']
-            return F.max(col).over(window)
-        else:
-            return F.max(col)
-
-    return compile_aggregator(t, expr, scope, fn, context, **kwargs)
+    return compile_aggregator(t, expr, scope, F.max, context, **kwargs)
 
 
 @compiles(ops.Min)
+@compiles(ops.CumulativeMin)
 def compile_min(t, expr, scope, context=None, **kwargs):
-    def fn(col):
-        if 'window' in kwargs:
-            window = kwargs['window']
-            return F.min(col).over(window)
-        else:
-            return F.min(col)
-
-    return compile_aggregator(t, expr, scope, fn, context, **kwargs)
+    return compile_aggregator(t, expr, scope, F.min, context, **kwargs)
 
 
 @compiles(ops.Mean)
+@compiles(ops.CumulativeMean)
 def compile_mean(t, expr, scope, context=None, **kwargs):
-    def fn(col):
-        if 'window' in kwargs:
-            window = kwargs['window']
-            return F.mean(col).over(window)
-        else:
-            return F.mean(col)
-
-    return compile_aggregator(t, expr, scope, fn, context, **kwargs)
+    return compile_aggregator(t, expr, scope, F.mean, context, **kwargs)
```

Comment:
@icexelloss could you provide a docstring here using numpy format? https://numpydoc.readthedocs.io/en/latest/format.html